b2sdk/transfer/emerge/emerger.py

```diff
@@ -130,14 +163,26 @@ def emerge_stream(
 
         :param int min_part_size: lower limit of part size for the transfer planner, in bytes
         :param int max_part_size: upper limit of part size for the transfer planner, in bytes
-
+        :param str,None large_file_sha1: result file SHA1 hash or ``None`` if not known
         """
         planner = self.get_emerge_planner(
             min_part_size=min_part_size,
             recommended_upload_part_size=recommended_upload_part_size,
             max_part_size=max_part_size,
         )
+
+        # ensure that write_intent_iterator is an iterator (and not just iterable, like a list)
+        if not isinstance(write_intent_iterator, Iterator):
+            write_intent_iterator = iter(write_intent_iterator)
+
+        first_write_intents, write_intent_iterator = iterator_peek(write_intent_iterator, 2)
+
         emerge_plan = planner.get_streaming_emerge_plan(write_intent_iterator)  # <--
+
+        file_info = self._get_updated_file_info_with_large_file_sha1(
```

Comment:
We should name the output differently. Otherwise we're updating reference to `file_info` which is the input of this function.